pkg/imgpkg/bundle/images_lock.go

```diff
@@ -39,45 +45,34 @@ func (o *ImagesLock) Merge(imgLock *ImagesLock) error {
 	return nil
 }
 
-func (o *ImagesLock) GenerateImagesLocations() error {
-	for i, imgRef := range o.ImageRefs() {
-		imageInBundleRepo, err := o.imageRelativeToBundle(imgRef.Image)
-		if err != nil {
-			return err
-		}
-
-		o.imagesLock.Images[i].AddLocation(imageInBundleRepo)
-	}
-	return nil
-}
-
 func (o *ImagesLock) AddImageRef(ref lockconfig.ImageRef) {
 	o.imagesLock.AddImageRef(ref)
 }
 
-// TODO: we should use LocationPrunedImageRefs as part of this function
 func (o *ImagesLock) LocalizeImagesLock() (lockconfig.ImagesLock, bool, error) {
 	var imageRefs []lockconfig.ImageRef
 	imagesLock := lockconfig.ImagesLock{
 		LockVersion: o.imagesLock.LockVersion,
 	}
 
+	skippedLocalization := false
 	for _, imgRef := range o.imagesLock.Images {
-		imageInBundleRepo, err := o.imageRelativeToBundle(imgRef.Image)
-		if err != nil {
-			return o.imagesLock, false, err
-		}
-
-		foundImg, err := o.checkImagesExist([]string{imageInBundleRepo, imgRef.Image})
+		foundImg, err := o.imgRetriever.FirstImageExists(imgRef.Locations())
 		if err != nil {
 			return o.imagesLock, false, err
 		}
 
 		// If cannot find the image in the bundle repo, will not localize any image
 		// We assume that the bundle was not copied to the bundle location,
 		// so there we cannot localize any image
-		if foundImg != imageInBundleRepo {
-			return o.imagesLock, true, nil
+		if foundImg != imgRef.PrimaryLocation() {
+			imageRefs = []lockconfig.ImageRef{}
+			// Remove the bundle location on all the Images
+			for _, image := range o.imagesLock.Images {
+				imageRefs = append(imageRefs, image.DiscardLocationsExcept(image.Image))
+			}
```

Comment:
nit: might be more readable if done outside of the loop below e.g.

```
if skippedLocalization {
   ...
}
```

just my 2c, not a blocking change.

Reply:
done